### PR TITLE
Migrate rest of {Cosmic,GlobalCosmic,Global,L2,L3,Standalone}MuonProduer to EventSetup consumes

### PR DESCRIPTION
**Pull request by makortel**
#### PR description:

This PR is part of #31061. It completes the EventSetup-consumes migration for bunch of Muon modules that were partly migrated earlier, but were causing failures in #31746.

#### PR validation:

Code compiles, these modules no longer triggers the exception of #31746 in a job using them.

### CODE REVIEW

**Comment by makortel** [comment]
@cmsbuild, please test

File: RecoMuon/TrackingTools/interface/MuonTrackFinder.h [hunk 1]
```diff
@@ -12,17 +12,15 @@
...
+#include "FWCore/Framework/interface/FrameworkfwdMostUsed.h"
 
 #include <vector>
 
-namespace edm {
-  class ParameterSet;
-  class Event;
-  class EventSetup;
-}  // namespace edm
 class TrackerTopology;
```
Comment (perrotta) [inline RecoMuon/TrackingTools/interface/MuonTrackFinder.h:22]: These two classes have been include'd above: no need to redefine them here
(and no need to have them still included in the MuonTrackFinder implementation file too)

Comment (makortel) [reply RecoMuon/TrackingTools/interface/MuonTrackFinder.h]: Thanks, removed.

**Comment by makortel** [comment]
@cmsbuild, please test

**Comment by perrotta** [comment]
+1
- Technical
- Jenkins tests pass

**Comment by Martin-Grunewald** [comment]
+1

**Comment by qliphy** [comment]
+1

**Pull request merged**